opennms-web-api/src/main/java/org/opennms/web/svclayer/support/DefaultGraphResultsService.java

```diff
@@ -154,6 +165,29 @@ public GraphResults findResults(ResourceId[] resourceIds, String[] reports, Stri
                 }
             }
         }
+        // GraphSelected case where all resources are fetched from generatedId
+        if (!Strings.isNullOrEmpty(generatedId) && m_jsonStore != null) {
+            Optional<String> result = m_jsonStore.get(generatedId, RESOURCE_IDS_CONTEXT);
+            if (result.isPresent()) {
+                try {
+                    String[] resourceArray = m_gson.fromJson(result.get(), String[].class);
+                    for (String resourceId : resourceArray) {
+                        try {
+                            //Resources are saved as keyValue resourceId=node[12312].
```

Comment:
Can you provide an example as to what the JSON content looks like? What does the content of the POST to the ReST endpoint look like?

Rather than splitting strings based on `=` it would probably better to send/store JSON objects and reference the fields we want by their key.

So this:
```
{
    [
        {"id": "a", "value": "node[123123]"},
        {"id": "b", "value": "node[456567]"}
    ]
}
```

instead of this (assuming this is what it currently looks like):
```
{
    [
        "a=node[123123]",
        "b=node[456567]"
    ]
}
```

Reply:
The values are just strings. `resourceId=node[3232]` is how servlet request parms were sent. I m using the same inputs here. I think it would be better just parse the values at front end before `POST` call and just send strings.

Reply:
Ok, I think the ReST endpoint should be updated to take a JSON array of IDs rather than key value pairs. It doesn't make sense to persist the whole key value pair when we only care about the ID portion.

The ReST endpoint's contract should be:
Input:
A JSON encoded list of resource IDs. i.e. `{["id1", "id2", "id3"]}`
Result:
A generated ID that maps all the given IDs.

Once that is done, we won't need to split the results anymore when we retrieve them from the JSON store. The client side will have to be smarter about how it sends the request however.

Also the list of resource IDs should be contained in the body of the `POST` message not as part of query parameter string (Assuming this is already the case, just making sure).

Reply:
Looks like you thought the same, seems the commit you just pushed does this 👍